internal/pkg/aead/aead.go

```diff
@@ -61,6 +67,11 @@ func (c *MiscreantCipher) Encrypt(plaintext []byte) (joined []byte, err error) {
 
 // Decrypt a value using AES-CMAC-SIV
 func (c *MiscreantCipher) Decrypt(joined []byte) ([]byte, error) {
+	aead, err := miscreant.NewAEAD(algorithmType, c.secret, miscreantNonceSize)
```

Comment:
It's worth pointing out that this seems to be a somewhat nontrivial initializer, which performs some block cipher encryption to derive its initial state, on each call.
https://github.com/miscreant/miscreant-go/blob/master/cmac/cmac.go#L45

My assumption is that `sso_proxy` is not particularly compute-bounded, but it could be interesting to throw this up in `stage`, and using Datadog to verify that we don't see an undue latency cost from these initializations.

Reply:
I suppose instead of instantiating new objects we could just lock around a single cipher?

I wonder which would be more performant.